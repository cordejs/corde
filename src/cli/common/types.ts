import { Command } from "commander";

export type ParamsFrom = "args" | "options" | "both";

export interface ICommandOptions {
  flags: string;
  description?: string;
  defaultValue?: string;
}

export interface ICliCommandConstructor {
  name?: string;
  program: Command;
  paramsFrom?: ParamsFrom;
}

export interface ICliCommand {
  readonly paramsFrom?: ParamsFrom;
  readonly command: Command;
  handler(...args: any[]): Promise<void> | void;
}

export interface ICliCommandConstructable {
  new (command: Command): ICliCommand;
}
