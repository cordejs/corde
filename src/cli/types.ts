export interface ICommand {
  action(...args: any[]): Promise<void> | void;
}
