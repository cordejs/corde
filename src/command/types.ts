import { ICordeBot } from "../types";

export interface ICreateMatcherParam {
  matcher?: string;
  isNot?: boolean;
  commandName?: string | boolean | number;
  channelId?: string;
  guildId?: string;
  isDebug?: boolean;
  isCascade?: boolean;
  cordeBot?: ICordeBot;
  trace?: string;
  utility?: boolean;
  mustSendCommand?: boolean;
}
