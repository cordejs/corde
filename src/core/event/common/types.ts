import { Client, ClientEvents } from "discord.js";

export type EventNames = keyof ClientEvents;

export type Spread<T extends EventNames> = ClientEvents[T][1] extends undefined
  ? ClientEvents[T] extends (infer U)[]
    ? U
    : never
  : ClientEvents[T];

export interface IDiscordEvent<TEvent extends EventNames, TFilter> {
  canListen(): boolean;
  getIntents(): string[];
  on(fn: (...args: ClientEvents[TEvent]) => void): this;
  once(filter?: TFilter): Promise<Spread<TEvent>>;
}

export interface IDiscordEventConstructable<TEvent extends EventNames, TFilter> {
  new (client: Client): IDiscordEvent<TEvent, TFilter>;
}

export type IntentType =
  | "GUILDS"
  | "GUILD_MEMBERS"
  | "GUILD_BANS"
  | "GUILD_EMOJIS_AND_STICKERS"
  | "GUILD_INTEGRATIONS"
  | "GUILD_WEBHOOKS"
  | "GUILD_INVITES"
  | "GUILD_VOICE_STATES"
  | "GUILD_PRESENCES"
  | "GUILD_MESSAGES"
  | "GUILD_MESSAGE_REACTIONS"
  | "GUILD_MESSAGE_TYPING"
  | "DIRECT_MESSAGES"
  | "DIRECT_MESSAGE_REACTIONS"
  | "DIRECT_MESSAGE_TYPING"
  | "GUILD_SCHEDULED_EVENTS";

export type EventClass = {
  [k in IntentType]: {
    bitfield: number;
    events: EventNames[];
  };
};
