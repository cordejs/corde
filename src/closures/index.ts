export * from "./group";
export * from "./test";
