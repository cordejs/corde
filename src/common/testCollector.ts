import { Queue } from "../data-structures";
import { IAssertionProps, IGroup, ITest, TestFunctionType, VoidLikeFunction } from "../types";

/**
 * Contain all information of data collected from files in runtime test
 * collection.
 * @internal
 */
class TestCollector {
  private constructor() {
    this.isInsideGroupClausure = false;
    this.isInsideTestClausure = false;
    this.groups = [];
    this.beforeEachFunctions = new Queue();
    this.afterAllFunctions = new Queue();
    this.beforeStartFunctions = new Queue();
    this.afterEachFunctions = new Queue();

    this.testClousureFunction = new Queue();
    this.groupClousureFunction = new Queue();

    this.tests = [];
    this.assertions = [];
    this.isolatedFunctions = [];
    this.testsFunctions = [];
    this.testsPass = 0;
    this.testsFailed = 0;
  }

  private static _instance: TestCollector;
  isInsideGroupClausure: boolean;
  isInsideTestClausure: boolean;

  assertions: IAssertionProps[];
  tests: ITest[];
  groups: IGroup[];

  beforeStartFunctions: Queue<VoidLikeFunction>;
  afterAllFunctions: Queue<VoidLikeFunction>;
  beforeEachFunctions: Queue<VoidLikeFunction>;

  afterEachFunctions: Queue<VoidLikeFunction>;
  testsFunctions: TestFunctionType[];
  isolatedFunctions: TestFunctionType[];

  testsPass: number;
  testsFailed: number;

  private testClousureFunction: Queue<VoidLikeFunction>;
  private groupClousureFunction: Queue<VoidLikeFunction>;

  static getInstance() {
    if (!TestCollector._instance) {
      TestCollector._instance = new TestCollector();
    }
    return TestCollector._instance;
  }

  addTestFunction(testFunction: TestFunctionType) {
    if (testFunction) {
      if (this.isInsideGroupClausure || this.isInsideTestClausure) {
        this.testsFunctions.push(testFunction);
      } else {
        this.isolatedFunctions.push(testFunction);
      }
    }
  }

  isInsideTestClausureFunctions() {
    return this.testsFunctions && this.testsFunctions.length > 0;
  }

  hasIsolatedTestFunctions() {
    return this.isolatedFunctions && this.isolatedFunctions.length > 0;
  }

  cloneTestFunctions() {
    return this.testsFunctions.map((m) => m);
  }

  cloneIsolatedTestFunctions() {
    return this.isolatedFunctions.map((m) => m);
  }

  clearTestFunctions() {
    this.testsFunctions = [];
  }

  clearIsolatedTestFunctions() {
    this.isolatedFunctions = [];
  }

  cleanAll() {
    this.tests = [];
    this.testsFunctions = [];
    this.groups = [];
  }

  addToGroupClousure(fn: () => void | Promise<void>) {
    this.groupClousureFunction.enqueue(fn);
  }

  async executeGroupClojure() {
    return await this.groupClousureFunction?.executeWithCatchCollectAsync();
  }

  addToTestClousure(fn: () => void | Promise<void>) {
    this.testClousureFunction.enqueue(fn);
  }

  async executeTestClojure() {
    return await this.testClousureFunction?.executeWithCatchCollectAsync();
  }
}

const testCollector = TestCollector.getInstance();
export { testCollector };
