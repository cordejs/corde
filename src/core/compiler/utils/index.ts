export * from "./import-js-in-directory";
export * from "./was-file-modified";
