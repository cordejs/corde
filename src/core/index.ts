export * from "./Group";
export * from "./TestFile";
export * from "./Config";
export * from "./debug";
export * from "./Events";
export * from "./injectGlobals";
export * from "./internalEvents";
export * from "./printHookError";
export * from "./Reader";
export * from "./runtime";
export * from "./summary";
export * from "./testCollector";
export * from "./TestExecutor";
export * from "./tsRegister";
export * from "./CordeBot";
