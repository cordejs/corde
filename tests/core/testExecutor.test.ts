import { TestExecutor } from "../../src/core/testExecutor";
import { LogUpdate } from "../../src/utils";
import { removeANSIColorStyle } from "../testHelper";

import { mockTimer } from "../mocks/mockTimer";
import { afterEach as _afterEach, beforeEach as _beforeEach } from "../../src/hooks";
import { IRunnerReport } from "../../src/types";
import { group, test } from "../../src/closures";
import { expect as _expect } from "../../src/expect";
import { testCollector } from "../../src/core/testCollector";

let logUpdate: LogUpdate;
let testRunner: TestExecutor;

mockTimer();

beforeEach(() => {
  logUpdate = new LogUpdate();
  testRunner = new TestExecutor(logUpdate);
  testCollector.clearTestFiles();
  testCollector.createTestFile("test file");
});

it("should print report for 1 test file, 1 test closure and 1 test function", async () => {
  test("test closure", () => {
    _expect(1).toEqual(1);
  });

  const report = await testRunner.runTestsAndPrint(testCollector.testFiles);
  expect(report).toEqual<IRunnerReport>({
    totalTests: 1,
    totalEmptyTestFiles: 0,
    totalEmptyTests: 0,
    totalTestFiles: 1,
    totalTestFilesFailed: 0,
    testTimer: "10ms",
    totalTestsFailed: 0,
    totalTestsPassed: 1,
    totalTestFilesPassed: 1,
  });
  expect(logUpdate.stdout).toMatchSnapshot();
  expect(removeANSIColorStyle(logUpdate.stdout)).toMatchSnapshot("without ANSI Colors");
});

it("should print report for 1 test file, 1 test closure and 2 test function", async () => {
  test("test closure", () => {
    _expect(1).toEqual(1);
    _expect(1).toEqual(1);
  });

  const report = await testRunner.runTestsAndPrint(testCollector.testFiles);
  expect(report).toEqual<IRunnerReport>({
    totalTests: 1,
    totalEmptyTestFiles: 0,
    totalEmptyTests: 0,
    totalTestFiles: 1,
    totalTestFilesFailed: 0,
    totalTestsFailed: 0,
    totalTestsPassed: 1,
    totalTestFilesPassed: 1,
    testTimer: "10ms",
  });
  expect(logUpdate.stdout).toMatchSnapshot();
  expect(removeANSIColorStyle(logUpdate.stdout)).toMatchSnapshot("without ANSI Colors");
});

it("should print a report for a group with a test with multiple expects", async () => {
  group("group", () => {
    test("test closure", () => {
      _expect(1).toEqual(1);
      _expect(1).toEqual(1);
    });
  });

  await testCollector.executeGroupClojure();
  const report = await testRunner.runTestsAndPrint(testCollector.testFiles);
  expect(report).toEqual<IRunnerReport>({
    totalTests: 1,
    totalEmptyTestFiles: 0,
    totalEmptyTests: 0,
    totalTestFiles: 1,
    totalTestFilesFailed: 0,
    totalTestsFailed: 0,
    totalTestsPassed: 1,
    totalTestFilesPassed: 1,
    testTimer: "10ms",
  });
  expect(logUpdate.stdout).toMatchSnapshot();
  expect(removeANSIColorStyle(logUpdate.stdout)).toMatchSnapshot("without ANSI Colors");
});

it("should print report for single group and a single test", async () => {
  group("group", () => {
    test("test closure", () => {
      _expect(1).not.toEqual(1);
    });
  });

  await testCollector.executeGroupClojure();
  const report = await testRunner.runTestsAndPrint(testCollector.testFiles);
  expect(report).toEqual<IRunnerReport>({
    totalTests: 1,
    totalEmptyTestFiles: 0,
    totalEmptyTests: 0,
    totalTestFiles: 1,
    totalTestFilesFailed: 1,
    totalTestFilesPassed: 0,
    totalTestsFailed: 1,
    totalTestsPassed: 0,
    testTimer: "10ms",
  });
  expect(logUpdate.stdout).toMatchSnapshot();
  expect(removeANSIColorStyle(logUpdate.stdout)).toMatchSnapshot("without ANSI Colors");
});

it("should get report for single group and a empty test", async () => {
  group("group", () => {
    test("test closure", () => {});
  });

  await testCollector.executeGroupClojure();
  const report = await testRunner.runTestsAndPrint(testCollector.testFiles);
  expect(report).toEqual<IRunnerReport>({
    totalTests: 1,
    totalEmptyTestFiles: 1,
    totalEmptyTests: 1,
    totalTestFiles: 1,
    totalTestFilesFailed: 0,
    totalTestsFailed: 0,
    totalTestsPassed: 0,
    totalTestFilesPassed: 0,
    testTimer: "10ms",
  });
  expect(logUpdate.stdout).toMatchSnapshot();
  expect(removeANSIColorStyle(logUpdate.stdout)).toMatchSnapshot("without ANSI Colors");
});

it("should get report for multiple tests inside a group", async () => {
  group("group", () => {
    test("test closure", () => {
      _expect(1).toEqual(1);
    });

    test("test closure", () => {
      _expect(1).toEqual(1);
    });

    test("test closure", () => {
      _expect(1).toEqual(1);
    });
  });

  await testCollector.executeGroupClojure();
  const report = await testRunner.runTestsAndPrint(testCollector.testFiles);
  expect(report).toEqual<IRunnerReport>({
    totalTests: 3,
    totalEmptyTestFiles: 0,
    totalEmptyTests: 0,
    totalTestFiles: 1,
    totalTestFilesFailed: 0,
    totalTestsFailed: 0,
    totalTestsPassed: 3,
    totalTestFilesPassed: 1,
    testTimer: "10ms",
  });
  expect(logUpdate.stdout).toMatchSnapshot();
  expect(removeANSIColorStyle(logUpdate.stdout)).toMatchSnapshot("without ANSI Colors");
});

it("should get report for multiple tests inside a group and failed tests", async () => {
  group("group", () => {
    test("test closure", () => {
      _expect(1).toEqual(1);
    });

    test("test closure", () => {
      _expect(1).toEqual(1);
    });

    test("test closure", () => {
      _expect(1).toEqual(1);
    });

    test("test closure", () => {
      _expect(1).not.toEqual(1);
    });
  });

  await testCollector.executeGroupClojure();
  const report = await testRunner.runTestsAndPrint(testCollector.testFiles);
  expect(report).toEqual<IRunnerReport>({
    totalTests: 4,
    totalEmptyTestFiles: 0,
    totalEmptyTests: 0,
    totalTestFiles: 1,
    totalTestFilesFailed: 1,
    totalTestsFailed: 1,
    totalTestsPassed: 3,
    totalTestFilesPassed: 0,
    testTimer: "10ms",
  });
  expect(logUpdate.stdout).toMatchSnapshot();
  expect(removeANSIColorStyle(logUpdate.stdout)).toMatchSnapshot("without ANSI Colors");
});

it("should get report for subgroups", async () => {
  group("g1", () => {
    group("g2", () => {
      group("g3", () => {
        group("g4", () => {
          group("g5", () => {
            group("g6", () => {
              test("test", () => {
                _expect(1).toEqual(1);
              });
            });
          });
        });
      });
    });
  });

  await testCollector.executeGroupClojure();
  const report = await testRunner.runTestsAndPrint(testCollector.testFiles);
  expect(report).toEqual<IRunnerReport>({
    totalTests: 1,
    totalEmptyTestFiles: 0,
    totalEmptyTests: 0,
    totalTestFiles: 1,
    totalTestFilesFailed: 0,
    totalTestsFailed: 0,
    totalTestsPassed: 1,
    totalTestFilesPassed: 1,
    testTimer: "10ms",
  });
  expect(logUpdate.stdout).toMatchSnapshot();
  expect(removeANSIColorStyle(logUpdate.stdout)).toMatchSnapshot("without ANSI Colors");
});

it("should get report for subtests", async () => {
  test("t1", () => {
    test("t2", () => {
      test("t3", () => {
        test("t4", () => {
          test("t5", () => {
            test("t6", () => {
              test("t7", () => {
                _expect(1).toEqual(1);
              });
            });
          });
        });
      });
    });
  });

  await testCollector.executeGroupClojure();
  const report = await testRunner.runTestsAndPrint(testCollector.testFiles);
  expect(report).toEqual<IRunnerReport>({
    totalTests: 7,
    totalEmptyTestFiles: 0,
    totalEmptyTests: 6,
    totalTestFiles: 1,
    totalTestFilesFailed: 0,
    totalTestsFailed: 0,
    totalTestsPassed: 1,
    totalTestFilesPassed: 1,
    testTimer: "10ms",
  });
  expect(logUpdate.stdout).toMatchSnapshot();
  expect(removeANSIColorStyle(logUpdate.stdout)).toMatchSnapshot("without ANSI Colors");
});

it("should get report for splited groups", async () => {
  group("group 1", () => {
    test("test closure 1", () => {
      _expect(1).toEqual(1);
    });

    test("test closure 2", () => {
      _expect(1).toEqual(1);
    });

    test("test closure 3", () => {
      _expect(1).toEqual(1);
    });

    test("test closure 4", () => {
      _expect(1).not.toEqual(1);
    });
  });

  group("group 2", () => {
    test("test closure 5", () => {
      _expect(1).toEqual(1);
    });

    test("test closure 6", () => {
      _expect(1).toEqual(1);
    });

    test("test closure 7", () => {
      _expect(1).toEqual(1);
    });

    test("test closure 8", () => {
      _expect(1).not.toEqual(1);
    });
  });

  group("group 3", () => {
    test("test closure 9", () => {
      _expect(1).toEqual(1);
    });

    test("test closure 10", () => {
      _expect(1).toEqual(1);
    });

    test("test closure 11", () => {
      _expect(1).toEqual(1);
    });

    test("test closure 12", () => {
      _expect(1).not.toEqual(1);
    });
  });

  await testCollector.executeGroupClojure();
  const report = await testRunner.runTestsAndPrint(testCollector.testFiles);
  expect(report).toEqual<IRunnerReport>({
    totalTests: 12,
    totalEmptyTestFiles: 0,
    totalEmptyTests: 0,
    totalTestFiles: 1,
    totalTestFilesFailed: 1,
    totalTestsFailed: 3,
    totalTestsPassed: 9,
    totalTestFilesPassed: 0,
    testTimer: "10ms",
  });
  expect(logUpdate.stdout).toMatchSnapshot();
  expect(removeANSIColorStyle(logUpdate.stdout)).toMatchSnapshot("without ANSI Colors");
});

it("should get report for multiple files", async () => {
  testCollector.clearTestFiles();
  testCollector.createTestFile("file 1");

  group("group 1", () => {
    test("test closure 1", () => {
      _expect(1).toEqual(1);
    });
  });

  await testCollector.executeGroupClojure();
  testCollector.createTestFile("file 2");

  group("group 2", () => {
    test("test closure 2", () => {
      _expect(1).toEqual(1);
    });
  });

  await testCollector.executeGroupClojure();
  testCollector.createTestFile("file 3");

  group("group 3", () => {
    test("test closure 3", () => {
      _expect(1).toEqual(1);
    });
  });

  await testCollector.executeGroupClojure();
  const report = await testRunner.runTestsAndPrint(testCollector.testFiles);
  expect(report).toEqual<IRunnerReport>({
    totalTests: 3,
    totalEmptyTestFiles: 0,
    totalEmptyTests: 0,
    totalTestFiles: 3,
    totalTestFilesFailed: 0,
    totalTestsFailed: 0,
    totalTestsPassed: 3,
    totalTestFilesPassed: 3,
    testTimer: "10ms",
  });
  expect(logUpdate.stdout).toMatchSnapshot();
  expect(removeANSIColorStyle(logUpdate.stdout)).toMatchSnapshot("without ANSI Colors");
});
