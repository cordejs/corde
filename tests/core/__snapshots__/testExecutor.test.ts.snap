// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`- 1`] = `
"
[41m[30m FAIL [39m[49m   [31mtest file[39m   [36m10ms[39m
   [32m[32m‚úî [39m[32m test closure 1[39m   [36m10ms[39m
   [32m[32m‚úî [39m[32m test closure 2[39m   [36m10ms[39m
   [32m[32m‚úî [39m[32m test closure 3[39m   [36m10ms[39m
   [31m[31mx[39m[31m test closure 4[39m   [36m10ms[39m
    
    Tip: toEqual asserts the value of properties. To compare their identity use [1mtoBe[22m
    
    expect([32mexpected[39m).not.toEqual([31mreceived[39m)
    
    expect: [32m1[39m !== [31m1[39m

    at toEqual /<fake>/<file>/<path>/testExecutor.test.ts:200:100
    at _internalTest (/<fake>/<file>/<path>/test.ts:200:100)
    at /<fake>/<file>/<path>/test.ts:200:100
    at /<fake>/<file>/<path>/executePromiseWithTimeout.ts:200:100
    at Object.action (/<fake>/<file>/<path>/test.ts:200:100)
    at TestExecutor.runTest (/<fake>/<file>/<path>/TestExecutor.ts:200:100)
    at process.processTicksAndRejections (node:internal/<fake>/<file>/<path>/task_queues:95:5)

   [32m[32m‚úî [39m[32m test closure 5[39m   [36m10ms[39m
   [32m[32m‚úî [39m[32m test closure 6[39m   [36m10ms[39m
   [32m[32m‚úî [39m[32m test closure 7[39m   [36m10ms[39m
   [31m[31mx[39m[31m test closure 8[39m   [36m10ms[39m
    
    Tip: toEqual asserts the value of properties. To compare their identity use [1mtoBe[22m
    
    expect([32mexpected[39m).not.toEqual([31mreceived[39m)
    
    expect: [32m1[39m !== [31m1[39m

    at toEqual /<fake>/<file>/<path>/testExecutor.test.ts:200:100
    at _internalTest (/<fake>/<file>/<path>/test.ts:200:100)
    at /<fake>/<file>/<path>/test.ts:200:100
    at /<fake>/<file>/<path>/executePromiseWithTimeout.ts:200:100
    at Object.action (/<fake>/<file>/<path>/test.ts:200:100)
    at TestExecutor.runTest (/<fake>/<file>/<path>/TestExecutor.ts:200:100)
    at process.processTicksAndRejections (node:internal/<fake>/<file>/<path>/task_queues:95:5)

   [32m[32m‚úî [39m[32m test closure 9[39m   [36m10ms[39m
   [32m[32m‚úî [39m[32m test closure 10[39m   [36m10ms[39m
   [32m[32m‚úî [39m[32m test closure 11[39m   [36m10ms[39m
   [31m[31mx[39m[31m test closure 12[39m   [36m10ms[39m
    
    Tip: toEqual asserts the value of properties. To compare their identity use [1mtoBe[22m
    
    expect([32mexpected[39m).not.toEqual([31mreceived[39m)
    
    expect: [32m1[39m !== [31m1[39m

    at toEqual /<fake>/<file>/<path>/testExecutor.test.ts:200:100
    at _internalTest (/<fake>/<file>/<path>/test.ts:200:100)
    at /<fake>/<file>/<path>/test.ts:200:100
    at /<fake>/<file>/<path>/executePromiseWithTimeout.ts:200:100
    at Object.action (/<fake>/<file>/<path>/test.ts:200:100)
    at TestExecutor.runTest (/<fake>/<file>/<path>/TestExecutor.ts:200:100)
    at process.processTicksAndRejections (node:internal/<fake>/<file>/<path>/task_queues:95:5)
"
`;

exports[`-: without ANSI Colors 1`] = `
"
 FAIL    test file   10ms
   ‚úî  test closure 1   10ms
   ‚úî  test closure 2   10ms
   ‚úî  test closure 3   10ms
   x test closure 4   10ms
    
    Tip: toEqual asserts the value of properties. To compare their identity use toBe
    
    expect(expected).not.toEqual(received)
    
    expect: 1 !== 1

    at toEqual /<fake>/<file>/<path>/testExecutor.test.ts:200:100
    at _internalTest (/<fake>/<file>/<path>/test.ts:200:100)
    at /<fake>/<file>/<path>/test.ts:200:100
    at /<fake>/<file>/<path>/executePromiseWithTimeout.ts:200:100
    at Object.action (/<fake>/<file>/<path>/test.ts:200:100)
    at TestExecutor.runTest (/<fake>/<file>/<path>/TestExecutor.ts:200:100)
    at process.processTicksAndRejections (node:internal/<fake>/<file>/<path>/task_queues:95:5)

   ‚úî  test closure 5   10ms
   ‚úî  test closure 6   10ms
   ‚úî  test closure 7   10ms
   x test closure 8   10ms
    
    Tip: toEqual asserts the value of properties. To compare their identity use toBe
    
    expect(expected).not.toEqual(received)
    
    expect: 1 !== 1

    at toEqual /<fake>/<file>/<path>/testExecutor.test.ts:200:100
    at _internalTest (/<fake>/<file>/<path>/test.ts:200:100)
    at /<fake>/<file>/<path>/test.ts:200:100
    at /<fake>/<file>/<path>/executePromiseWithTimeout.ts:200:100
    at Object.action (/<fake>/<file>/<path>/test.ts:200:100)
    at TestExecutor.runTest (/<fake>/<file>/<path>/TestExecutor.ts:200:100)
    at process.processTicksAndRejections (node:internal/<fake>/<file>/<path>/task_queues:95:5)

   ‚úî  test closure 9   10ms
   ‚úî  test closure 10   10ms
   ‚úî  test closure 11   10ms
   x test closure 12   10ms
    
    Tip: toEqual asserts the value of properties. To compare their identity use toBe
    
    expect(expected).not.toEqual(received)
    
    expect: 1 !== 1

    at toEqual /<fake>/<file>/<path>/testExecutor.test.ts:200:100
    at _internalTest (/<fake>/<file>/<path>/test.ts:200:100)
    at /<fake>/<file>/<path>/test.ts:200:100
    at /<fake>/<file>/<path>/executePromiseWithTimeout.ts:200:100
    at Object.action (/<fake>/<file>/<path>/test.ts:200:100)
    at TestExecutor.runTest (/<fake>/<file>/<path>/TestExecutor.ts:200:100)
    at process.processTicksAndRejections (node:internal/<fake>/<file>/<path>/task_queues:95:5)
"
`;

exports[`should get report for multiple files 1`] = `
"
[42m[30m PASS [39m[49m   file 1   [36m10ms[39m
   [32m[32m‚úî [39m[32m test closure 1[39m   [36m10ms[39m
[42m[30m PASS [39m[49m   file 2   [36m10ms[39m
   [32m[32m‚úî [39m[32m test closure 2[39m   [36m10ms[39m
[42m[30m PASS [39m[49m   file 3   [36m10ms[39m
   [32m[32m‚úî [39m[32m test closure 3[39m   [36m10ms[39m"
`;

exports[`should get report for multiple files: without ANSI Colors 1`] = `
"
 PASS    file 1   10ms
   ‚úî  test closure 1   10ms
 PASS    file 2   10ms
   ‚úî  test closure 2   10ms
 PASS    file 3   10ms
   ‚úî  test closure 3   10ms"
`;

exports[`should get report for multiple tests inside a group 1`] = `
"
[42m[30m PASS [39m[49m   test file   [36m10ms[39m
   [32m[32m‚úî [39m[32m test closure[39m   [36m10ms[39m
   [32m[32m‚úî [39m[32m test closure[39m   [36m10ms[39m
   [32m[32m‚úî [39m[32m test closure[39m   [36m10ms[39m"
`;

exports[`should get report for multiple tests inside a group and failed tests 1`] = `
"
[41m[30m FAIL [39m[49m   [31mtest file[39m   [36m10ms[39m
   [32m[32m‚úî [39m[32m test closure[39m   [36m10ms[39m
   [32m[32m‚úî [39m[32m test closure[39m   [36m10ms[39m
   [32m[32m‚úî [39m[32m test closure[39m   [36m10ms[39m
   [31m[31mx[39m[31m test closure[39m   [36m10ms[39m
    
    Tip: toEqual asserts the value of properties. To compare their identity use [1mtoBe[22m
    
    expect([32mexpected[39m).not.toEqual([31mreceived[39m)
    
    expect: [32m1[39m !== [31m1[39m

    at toEqual /<fake>/<file>/<path>/testExecutor.test.ts:200:100
    at _internalTest (/<fake>/<file>/<path>/test.ts:200:100)
    at /<fake>/<file>/<path>/test.ts:200:100
    at /<fake>/<file>/<path>/executePromiseWithTimeout.ts:200:100
    at Object.action (/<fake>/<file>/<path>/test.ts:200:100)
    at TestExecutor.runTest (/<fake>/<file>/<path>/TestExecutor.ts:200:100)
    at process.processTicksAndRejections (node:internal/<fake>/<file>/<path>/task_queues:95:5)
"
`;

exports[`should get report for multiple tests inside a group and failed tests: without ANSI Colors 1`] = `
"
 FAIL    test file   10ms
   ‚úî  test closure   10ms
   ‚úî  test closure   10ms
   ‚úî  test closure   10ms
   x test closure   10ms
    
    Tip: toEqual asserts the value of properties. To compare their identity use toBe
    
    expect(expected).not.toEqual(received)
    
    expect: 1 !== 1

    at toEqual /<fake>/<file>/<path>/testExecutor.test.ts:200:100
    at _internalTest (/<fake>/<file>/<path>/test.ts:200:100)
    at /<fake>/<file>/<path>/test.ts:200:100
    at /<fake>/<file>/<path>/executePromiseWithTimeout.ts:200:100
    at Object.action (/<fake>/<file>/<path>/test.ts:200:100)
    at TestExecutor.runTest (/<fake>/<file>/<path>/TestExecutor.ts:200:100)
    at process.processTicksAndRejections (node:internal/<fake>/<file>/<path>/task_queues:95:5)
"
`;

exports[`should get report for multiple tests inside a group: without ANSI Colors 1`] = `
"
 PASS    test file   10ms
   ‚úî  test closure   10ms
   ‚úî  test closure   10ms
   ‚úî  test closure   10ms"
`;

exports[`should get report for single group and a empty test 1`] = `
"
[43m[30m EMPTY [39m[49m  test file   [36m10ms[39m
   [93m ‚óè test closure (empty)[39m   [36m10ms[39m"
`;

exports[`should get report for single group and a empty test: without ANSI Colors 1`] = `
"
 EMPTY   test file   10ms
    ‚óè test closure (empty)   10ms"
`;

exports[`should get report for subgroups 1`] = `
"
[42m[30m PASS [39m[49m   test file   [36m10ms[39m
   [32m[32m‚úî [39m[32m test[39m   [36m10ms[39m"
`;

exports[`should get report for subgroups: without ANSI Colors 1`] = `
"
 PASS    test file   10ms
   ‚úî  test   10ms"
`;

exports[`should get report for subtests 1`] = `
"
[42m[30m PASS [39m[49m   test file   [36m10ms[39m
   [93m ‚óè t1 (empty)[39m   [36m10ms[39m
   [93m ‚óè t2 (empty)[39m   [36m10ms[39m
   [93m ‚óè t3 (empty)[39m   [36m10ms[39m
   [93m ‚óè t4 (empty)[39m   [36m10ms[39m
   [93m ‚óè t5 (empty)[39m   [36m10ms[39m
   [93m ‚óè t6 (empty)[39m   [36m10ms[39m
   [32m[32m‚úî [39m[32m t7[39m   [36m10ms[39m"
`;

exports[`should get report for subtests: without ANSI Colors 1`] = `
"
 PASS    test file   10ms
    ‚óè t1 (empty)   10ms
    ‚óè t2 (empty)   10ms
    ‚óè t3 (empty)   10ms
    ‚óè t4 (empty)   10ms
    ‚óè t5 (empty)   10ms
    ‚óè t6 (empty)   10ms
   ‚úî  t7   10ms"
`;

exports[`should print a report for a group with a test with multiple expects 1`] = `
"
[42m[30m PASS [39m[49m   test file   [36m10ms[39m
   [32m[32m‚úî [39m[32m test closure[39m   [36m10ms[39m"
`;

exports[`should print a report for a group with a test with multiple expects: without ANSI Colors 1`] = `
"
 PASS    test file   10ms
   ‚úî  test closure   10ms"
`;

exports[`should print report for 1 test file, 1 test closure and 1 test function 1`] = `
"
[42m[30m PASS [39m[49m   test file   [36m10ms[39m
   [32m[32m‚úî [39m[32m test closure[39m   [36m10ms[39m"
`;

exports[`should print report for 1 test file, 1 test closure and 1 test function: without ANSI Colors 1`] = `
"
 PASS    test file   10ms
   ‚úî  test closure   10ms"
`;

exports[`should print report for 1 test file, 1 test closure and 2 test function 1`] = `
"
[42m[30m PASS [39m[49m   test file   [36m10ms[39m
   [32m[32m‚úî [39m[32m test closure[39m   [36m10ms[39m"
`;

exports[`should print report for 1 test file, 1 test closure and 2 test function: without ANSI Colors 1`] = `
"
 PASS    test file   10ms
   ‚úî  test closure   10ms"
`;

exports[`should print report for single group and a single test 1`] = `
"
[41m[30m FAIL [39m[49m   [31mtest file[39m   [36m10ms[39m
   [31m[31mx[39m[31m test closure[39m   [36m10ms[39m
    
    Tip: toEqual asserts the value of properties. To compare their identity use [1mtoBe[22m
    
    expect([32mexpected[39m).not.toEqual([31mreceived[39m)
    
    expect: [32m1[39m !== [31m1[39m

    at toEqual /<fake>/<file>/<path>/testExecutor.test.ts:200:100
    at _internalTest (/<fake>/<file>/<path>/test.ts:200:100)
    at /<fake>/<file>/<path>/test.ts:200:100
    at /<fake>/<file>/<path>/executePromiseWithTimeout.ts:200:100
    at Object.action (/<fake>/<file>/<path>/test.ts:200:100)
    at TestExecutor.runTest (/<fake>/<file>/<path>/TestExecutor.ts:200:100)
    at process.processTicksAndRejections (node:internal/<fake>/<file>/<path>/task_queues:95:5)
"
`;

exports[`should print report for single group and a single test: without ANSI Colors 1`] = `
"
 FAIL    test file   10ms
   x test closure   10ms
    
    Tip: toEqual asserts the value of properties. To compare their identity use toBe
    
    expect(expected).not.toEqual(received)
    
    expect: 1 !== 1

    at toEqual /<fake>/<file>/<path>/testExecutor.test.ts:200:100
    at _internalTest (/<fake>/<file>/<path>/test.ts:200:100)
    at /<fake>/<file>/<path>/test.ts:200:100
    at /<fake>/<file>/<path>/executePromiseWithTimeout.ts:200:100
    at Object.action (/<fake>/<file>/<path>/test.ts:200:100)
    at TestExecutor.runTest (/<fake>/<file>/<path>/TestExecutor.ts:200:100)
    at process.processTicksAndRejections (node:internal/<fake>/<file>/<path>/task_queues:95:5)
"
`;
